### ISSUE TITLE
Task List 29/04/2019

### ISSUE THREAD

**Issue by cseale** [cseale/fix-it-in-post.net#14]
End Goal: Recreate https://nl.mathworks.com/help/deeplearning/examples/denoise-speech-using-deep-learning-networks.html

- [x] : Create small sample of preprocessed data to validate pipeline (< 10 - 100 samples?)
- [x] : Create Loss Function
- [x] : Build Fully Connected Neural Network Architecture
- [x] : Train simple network and reconstruct audio

**Comment by GabrieleMazzola** [comment]
Good resource for STFT: http://research.cs.tamu.edu/prism/lectures/sp/l6.pdf

**Comment by esilezz** [comment]
SNR: 
- https://dsp.stackexchange.com/questions/14862/measure-the-snr-of-a-signal-in-the-frequency-domain --> procedure to approximate
-  Approximate procedure --> Get an FFT,  Sum power in all bins. This is total power. Next, zero all bins with signal-related harmonics (and a few bins on either side),.  Sum all bins again.  This is "noise" (non-signal) power. (Total power - noise power)/noise power = SNR
- estimation of the noise: http://cdsads.u-strasbg.fr/abs/2018A%26A...609A..39C
- approximation with the beta sigma procedure (IMPLEMENTED) --> https://www.hs.uni-hamburg.de/DE/Ins/Per/Czesla/PyA/PyA/pyaslDoc/aslDoc/estimateSNR.html

Other metrics we could possibly use:
- Total Harmonic Distortion (THD) --> used to check how non linear is a system
